analysis/study_definition_wave1.py

```diff
@@ -779,6 +779,48 @@
         on_or_before="index_date",
         find_last_match_in_period=True,
     ),
+    # Date of first COVID vaccination - source nhs-covid-vaccination-coverage
+    covid_vax_date_1=patients.with_tpp_vaccination_record(
+        target_disease_matches="SARS-2 CORONAVIRUS",
+        between=["2020-12-01", "index_date"],  # any dose recorded after 01/12/2020
+        find_first_match_in_period=True,
+        returning="date",
+        date_format="YYYY-MM-DD",
+        return_expectations={
+            "date": {"earliest": "2020-12-01", "latest": "index_date"},
+            "incidence": 0.8,
+        },
+    ),
+    # Date of second COVID vaccination - source nhs-covid-vaccination-coverage
+    covid_vax_date_2=patients.with_tpp_vaccination_record(
+        target_disease_matches="SARS-2 CORONAVIRUS",
+        between=["covid_vax_date_1 + 1 day", "index_date"],  # from day after previous dose
+        find_first_match_in_period=True,
+        returning="date",
+        date_format="YYYY-MM-DD",
+        return_expectations={
+            "date": {"earliest": "2020-12-01", "latest": "index_date"},
+            "incidence": 0.6,
+        },
+    ),
+    # Date of third COVID vaccination (primary or booster) -
```

Comment:
Since the third wave baseline (28 May 2021) predates the first booster campaigns (Sep 2021), covid_vax_date_3 should be 0 by definition in all waves. But no harm in keeping in as it will be ready to go if we want to extend to further waves after booster roll-out.

Reply:
Ah, good one. I'll leave like this now, but will check whether it is null when viewing data summaries.